source/l4_testing/can/can_test.py

```diff
@@ -0,0 +1,37 @@
+import can
+import usb
+
+dev = usb.core.find(idVendor=0x1D50, idProduct=0x606F)
+channel = dev.product
+bus_num = dev.bus
+addr = dev.address
+del(dev)
+bus = can.Bus(bustype="gs_usb", channel=channel, bus=bus_num, address=addr, bitrate=500000)
+
+msg = can.Message(arbitration_id=0x1400028b,
+                  data=[0xC5, 0x1A, 0, 1],
+                  is_extended_id=True)
+
+msg2 = can.Message(arbitration_id=0x1400028a,
+                  data=[5, 2, 3, 4],
+                  is_extended_id=True)
+
+msg3 = can.Message(arbitration_id=0x1400008c,
+                  data=[0b00100101],
+                  is_extended_id=True)
+
+# bus.send_periodic(msg, 0.2)
+# bus.send_periodic(msg2, 0.1)
+bus.send_periodic(msg3, 0.5)
+
+try:
+    while(True):
+        rx = bus.recv(3)
+        if(rx.is_error_frame):
+            print("Error frame: " + str(rx.error_state_indicator))
+        
+        print(rx)
```

Comment:
In the future it would be cool to use the generated DBC file and have cantools print out the decoded signals as well.

Reply:
Yes, for sure. I will be relying on cantools for the message decoding as much as possible within the dashboard application. Also, if we ever need to dump can frames to a log file, it will be quite simple to dump them pre-decoded.